format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -41,7 +41,12 @@
    */
   public static void detectDataTypeOfRowValues(Map<String, Schema> override,
                                                DataTypeDetectorStatusKeeper dataTypeDetectorStatusKeeper,
-                                               String[] columnNames, String[] rowValues) {
+                                               String[] columnNames, String[] rowValues) throws RuntimeException {
+    if (rowValues.length < columnNames.length) {
+      throw new RuntimeException(String.format(
```

Comment:
could we just add this to failure collector instead?

Reply:
Changed. The getSchema and getDefaultSchema were present in DelimitedConfig.java instead of using them to override functions in DelimitedConfigProvider.java. As a result it was not easy to get access to format context or failure collector. I had to move these functions over. I referred the style followed in [format-avro](https://github.com/cdapio/hydrator-plugins/tree/develop/format-avro/src/main/java/io/cdap/plugin/format/avro/input) while doing this.